ITR/data/excel.py

```diff
@@ -39,106 +29,167 @@ def _check_sector_data(self) -> None:
+
+        benchmark_projection = benchmark_projection.loc[list(zip(sectors, regions)),
+                                                        range(TemperatureScoreConfig.CONTROLS_CONFIG.base_year,
+                                                              TemperatureScoreConfig.CONTROLS_CONFIG.target_end_year + 1)]
+        benchmark_projection.index = sectors.index
+
+        return benchmark_projection
+
+
+class ExcelProviderIntensityBenchmark(IntensityBenchmarkDataProvider):
+    def __init__(self, excel_path: str, benchmark_temperature: float,
+                 benchmark_global_budget: float, AFOLU_included: bool,
+                 column_config: Type[ColumnsConfig] = ColumnsConfig,
+                 tempscore_config: Type[TemperatureScoreConfig] = TemperatureScoreConfig):
+        super().__init__(benchmark_temperature, benchmark_global_budget, AFOLU_included)
+        self.benchmark_excel = pd.read_excel(excel_path, sheet_name=None, skiprows=0)
+        self._check_sector_data()
+        self.temp_config = tempscore_config
+        self.column_config = column_config
+
+    def get_SDA_intensity_benchmarks(self, company_info_at_base_year: pd.DataFrame) -> pd.DataFrame:
+        """
+        Overrides subclass method
+        returns a Dataframe with intensity benchmarks per company_id given a region and sector.
+        :param company_info_at_base_year: DataFrame with at least the following columns :
+        ColumnsConfig.COMPANY_ID, ColumnsConfig.BASE_EI ColumnsConfig.SECTOR and ColumnsConfig.REGION
+        :return: A DataFrame with company and SDA intensity benchmarks per calendar year per row
+        """
+        intensity_benchmarks = self._get_intensity_benchmarks(company_info_at_base_year)
+        decarbonization_paths = self._get_decarbonizations_paths(intensity_benchmarks)
+        last_ei = intensity_benchmarks[self.temp_config.CONTROLS_CONFIG.target_end_year]
+        ei_base = company_info_at_base_year[self.column_config.BASE_EI]
+
+        return decarbonization_paths.mul((ei_base - last_ei), axis=0).add(last_ei, axis=0)
 
-        :param df_targets: pandas Dataframe with targets
-        :return: A list containing the targets
+    def _get_decarbonizations_paths(self, intensity_benchmarks: pd.DataFrame) -> pd.DataFrame:
         """
-        logger = logging.getLogger(__name__)
-        targets = df_targets.to_dict(orient="records")
-        model_targets: List[IDataProviderTarget] = []
-        for target in targets:
-            try:
-                model_targets.append(IDataProviderTarget.parse_obj(target))
-            except ValidationError as e:
-                logger.warning(
-                    "(one of) the target(s) of company %s is invalid and will be skipped" % target[self.c.COMPANY_NAME])
-                pass
-        return model_targets
+        Overrides subclass method
+        returns a Dataframe with intensity benchmarks per company_id given a region and sector.
```

Comment:
Change comment. Suggestion: Returns a DataFrame with the projected decarbonization paths for the supplied companies in intensity_benchmarks.